backend/infrahub/core/merge.py

```diff
@@ -434,12 +434,10 @@ async def merge_graph(  # pylint: disable=too-many-branches,too-many-statements
 
     async def merge_repositories(self) -> None:
         # Collect all Repositories in Main because we'll need the commit in Main for each one.
-        repos_in_main_list: list[CoreGenericRepository] = await NodeManager.query(
-            schema=InfrahubKind.REPOSITORY, db=self.db
-        )
+        repos_in_main_list: list[CoreRepository] = await NodeManager.query(schema=InfrahubKind.REPOSITORY, db=self.db)
```

Comment:
this should work now, 
```suggestion
        repos_in_main_list = await NodeManager.query(schema=CoreRepository, db=self.db)
```

Reply:
Ah yes good catch. I was only thinking that the old helper was incorrect and focused on fixing that so I didn't consider that we had better options now.